addons/website_version/models/experiment.py

```diff
@@ -0,0 +1,300 @@
+        if not website_id:
+            raise Warning("You must specify the website.")
+        for goal in gm_obj.get_goal_info(website_id)[1]['items']:
+            if not goals_obj.search([('name', '=', goal['name'])]):
+                vals = {'name': goal['name'], 'google_ref': 'ga:goal' + goal['id'] + 'Completions'}
+                goals_obj.create(vals)
+
+    @api.model
+    def check_no_overlap(self, version_ids):
+        #Check if version_ids don't overlap with running experiments and return the name of the experiment if there 's an overlap
+        version_keys = set([v['key'] for v in self.env['ir.ui.view'].search_read([('version_id', 'in', version_ids)], ['key'])])
+        exp_mod = self.env['website_version.experiment']
+        exps = exp_mod.search([('state', '=', 'running'), ('website_id', '=', self.env.context.get('website_id'))])
+        for exp in exps:
+            for exp_ver in exp.experiment_version_ids:
+                for view in exp_ver.version_id.view_ids:
+                    if view.key in version_keys:
+                        return (False, exp.name)
+        return (True, "")
+
+
+class google_management(models.AbstractModel):
+
+    STR_SERVICE = 'management'
+
+    _name = 'google.%s' % STR_SERVICE
+
+    @api.model
+    def generate_data(self, experiment, isCreating=False):
+        
+        data = {
+            'name': experiment['name'],
+            'status': experiment['status'],
+            'variations': experiment['variations']
+        }
+
+        return data
+
+    @api.model
+    def create_an_experiment(self, data, website_id):
+        gs_pool = self.env['google.service']
+        website = self.env['website'].browse(website_id)[0]
+        webPropertyId = website.google_analytics_key
+        action_id = self.env['ir.model.data'].xmlid_to_res_id('website_version.action_website_view')
+        if not webPropertyId:
+            raise exceptions.RedirectWarning('Click on the website you want to make A/B testing and configure the Google Analytics Key and View ID', action_id, 'go to the websites menu')
+        accountId = webPropertyId.split('-')[1]
+        profileId = website.google_analytics_view_id
+        if not profileId:
+            raise exceptions.RedirectWarning('Click on the website you want to make A/B testing and configure the Google Analytics Key and View ID', action_id, 'go to the websites menu')
+        url = '/analytics/v3/management/accounts/%s/webproperties/%s/profiles/%s/experiments?access_token=%s' % (accountId, webPropertyId, profileId, self.get_token())
+        headers = {'Content-type': 'application/json', 'Accept': 'text/plain'}
+        data_json = simplejson.dumps(data)
+        try:
+            x = gs_pool._do_request(url, data_json, headers, type='POST')
+            result = x[1]['id']
+        except Exception, e:
+            _logger.info(_('An exception occured during the google analytics rquest: %s') % e)
```

Comment:
I suppose this is required ? Having an argument "raise" allowing to return False or raise could'nt help ?

By the way, should'nt we maybe differentiate the exceptions, if feasible ? Like credentials issues / connection issue / other issue ?